core/math/src/main/java/org/arakhne/afc/math/geometry/d2/afp/BasicPathShadow2afp.java

```diff
@@ -209,83 +207,36 @@ private void discretizePathIterator(
-                        endx, endy);
-                discretizePathIterator(
-                        localPath.getPathIterator(MathConstants.SPLINE_APPROXIMATION_RATIO),
-                        x1, y1, x2, y2);
-                if (this.crossings == MathConstants.SHAPE_INTERSECTS) {
-                    return;
-                }
+                setQuadToFromDiscretizePathIterator(element, pi, x1, y1, x2, y2, curx, cury, endx, endy);
                 curx = endx;
                 cury = endy;
                 break;
             case CURVE_TO:
                 endx = element.getToX();
                 endy = element.getToY();
                 // only for local use.
-                localPath = pi.getGeomFactory().newPath(pi.getWindingRule());
-                localPath.moveTo(curx, cury);
-                localPath.curveTo(
-                        element.getCtrlX1(), element.getCtrlY1(),
-                        element.getCtrlX2(), element.getCtrlY2(),
-                        endx, endy);
-                discretizePathIterator(
-                        localPath.getPathIterator(MathConstants.SPLINE_APPROXIMATION_RATIO),
-                        x1, y1, x2, y2);
-                if (this.crossings == MathConstants.SHAPE_INTERSECTS) {
-                    return;
-                }
+                setCurveToFromDiscretizePathIterator(element, pi, x1, y1, x2, y2, curx, cury, endx, endy);
                 curx = endx;
                 cury = endy;
                 break;
             case ARC_TO:
                 endx = element.getToX();
                 endy = element.getToY();
                 // only for local use.
-                localPath = pi.getGeomFactory().newPath(pi.getWindingRule());
-                localPath.moveTo(curx, cury);
-                localPath.arcTo(
-                        endx, endy,
-                        element.getRadiusX(), element.getRadiusY(),
-                        element.getRotationX(), element.getLargeArcFlag(),
-                        element.getSweepFlag());
-                discretizePathIterator(
-                        localPath.getPathIterator(MathConstants.SPLINE_APPROXIMATION_RATIO),
-                        x1, y1, x2, y2);
-                if (this.crossings == MathConstants.SHAPE_INTERSECTS) {
-                    return;
-                }
+                setArcToFromDiscretizePathIterator(element, pi, x1, y1, x2, y2, curx, cury, endx, endy);
                 curx = endx;
                 cury = endy;
                 break;
             case CLOSE:
-                if (cury != movy || curx != movx) {
-                    crossSegmentTwoShadowLines(
-                            curx, cury,
-                            movx, movy,
-                            x1, y1, x2, y2);
-                }
-                if (this.crossings != 0) {
-                    return;
-                }
+                setCloseFromDiscretizePathIterator(x1, y1, x2, y2, curx, cury, movx, movy);
```

Comment:
The replaced code was supposed to exit this function if crossings is not zero. The replacement function does not provide the same feature.